design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,170 @@
+
+### Scenario 4a: The option for record preview is enabled
+
+#### Steps for 4a
+
+- The user opens a table that contains columns with a foreign key constraint applied.
+  - The column contains the referenced values for each cell. The first five fields are displayed.
+    - The displayed fields are presented as concatenated values. Visual contrast is created between the field name and its corresponding value to be easier to read.
+    - When clicked, the field displays the [record selector](#record-selector) component containing a pre-filtered list. A single matching item is displayed.
+
+### Scenario 4b: The option for record preview is disabled
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains the primary key value for the referenced column only.
+  - When clicked, the field displays the [record selector](#record-selector) component containing a pre-filtered list. A single matching item is displayed.
+
+#### Steps for 4b
+
+## Scenario 5: User Edits the Values of a Column With a Foreign Key Constraint Applied
+
+Linked records can be changed or removed according to the user's preference. Since values are references to records in other tables, users must understand that changes to references don't affect the original records. However, changes to the values of records in the referenced table do.
+
+### Scenario 5a: The field has an existing value
+
+#### Steps for 5a
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains the referenced values for each cell.
+  - When clicked, a pre-filtered list is displayed inside the [record selector](#record-selector) component. The dropdown includes a search input, that if modified, will show other matching records. Clicking on any of those records will close the dropdown and replace the existing one.
+  - Note that in the case of multi-column foreign key constraints, the system could modify additional fields.
+
+### Scenario 5b: The field is empty
+
+#### Steps for 5b
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains no values.
+  - When clicked, a list of records is displayed inside the [record selector](#record-selector) component. The list shows the first 50 records.
+  - An text input field is available within the dropdown to filter the records based on partial or complete value matches. As the user types in a value, the list narrows down the records. The system can search only the first five fields. Search across all types is supported.
+  - If there's a single match for the entered value, it should become highlighted, allowing the user to confirm the selection.
+
+### Scenario 5 Questions
+
+- What if the PK field is not one of the first five fields?
```

Comment:
There could be an option to do a search across all columns(wouldn't want to show this only if no rows are found as the returned row can be something the user is not looking for, but not sure how to restrict them from not using it often).

Having the user toggle the columns to search for can be another alternate

Reply:
For now, the idea will be to point users towards changing the search columns if no results are found. I think your concern about a 'search across all' might encourage the behavior you describe, which might not be ideal for performance.